services/api/db/serializers.py

```diff
@@ -0,0 +1,132 @@
+from rest_framework import serializers
+from db.models import User
+from django.utils.encoding import smart_str, force_bytes, DjangoUnicodeDecodeError
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.core.validators import validate_email
+from django.core.exceptions import ValidationError
+
+#Method for validating password.
+def validate_confirm_password(password,confirm_password):
+    if password != confirm_password:
+        return False
+    return True
+
+class SignUpEndPointSerializer(serializers.ModelSerializer):
+    confirm_password = serializers.CharField(style={'input_type': 'password'}, write_only=True)
+    class Meta:
+        model = User
+        fields= ['email', 'name', 'password', 'confirm_password']
+        extra_kwargs={
+            'password': {'write_only': True}
+        }
+    #To validate whether the entered email and password are correct.
+    def validate(self, data):
+        email = data.get('email')
+        if not email:
+            raise ValueError("Email field is required")
+        password, confirm_password = data.get('password'), data.get('confirm_password')
+        if not password or not confirm_password:
+            raise ValueError("Either password or confirm_password is empty")
+        try:
+            validate_email(email)
+        except:
+            raise serializers.ValidationError("Please provide a valid email address.")
+        if validate_confirm_password(password,confirm_password):
+            return data
+        else:
+            raise serializers.ValidationError("Password and confirm password don't match.")
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+        
+class SignInEndPointSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(max_length=255,
+                style={'input_type': 'email', 'placeholder': 'Email Address'})
+    password = serializers.CharField(max_length=255,
+                 style={'input_type' : 'password'}, write_only=True)
+    class Meta:
+        model = User
+        fields = ['email', 'password']
+
+class UserProfileSerialiser(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+    #New password type could be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
```

Comment:
```suggestion
    password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
```
is the style ={'input_type': 'password'} required in the serialializer?